### DISCUSSION TITLE
Direct file copies between nodes without network cost of going through proxy

### DISCUSSION THREAD

**Discussion by pieter-lautus** [gravitational/teleport#33285] [Q&A] [answered]
We're currently evaluating Teleport Cloud. One wrinkle is that we are in South Africa and the Teleport Cloud proxies are in the US, so SSH connections will pay the penalty of a round trip through the Atlantic.

At certain points of our software lifecycle we need to transfer backup files that are hundreds of gigabytes between servers, and we need to do it ASAP, so the extra transatlantic network hop will not be acceptable.

What would be the secure way of supporting this within Teleport?

One option would be for us to run our own Teleport proxies in the same datacenter private network that we are copying from. Would this count as a whole new Teleport cluster we have to join to our Cloud cluster? Does the license allow this?

Or can we run our own additional proxies on our own hardware and make them part of the Cloud cluster? I can't find anything in the documentation on how to install and run a proxy.

Or is this a scenario that calls for running Teleport in standard mode instead of reverse tunnel mode?

**Comment by webvictim** [comment, marked as answer]
Hey Pieter - it's not possible to join your own proxies to a Teleport Cloud cluster. It's also not possible to join nodes without using reverse tunnels when using Teleport Cloud.

Teleport Cloud does deploy proxies in all the regions listed here: https://goteleport.com/docs/choose-an-edition/teleport-cloud/architecture/#proxies - as such you shouldn't be seeing transatlantic latency when copying files via Teleport, connections should be routed through the (slightly) more local `eu-central-1` region. Admittedly, none of those proxies are especially close to South Africa, so the added latency may still be unacceptable for you.

In this situation the only way to reduce the latency would be to use a self-hosted Teleport deployment instead which would allow you to deploy your own proxies locally. You could technically configure this as a "leaf cluster" which you could join to a "root" Teleport Cloud cluster and federate logins/audit logs etc (see docs for details [[1](https://goteleport.com/docs/architecture/trustedclusters/)] [[2](https://goteleport.com/docs/management/admin/trustedclusters/)])

If you'd require an Enterprise license, you'd need to speak to a Teleport Account Executive about your use case to determine whether it'd be possible. If you don't have an Account Executive assigned already, you can email sales at goteleport dot com with details to get in contact.

1: https://goteleport.com/docs/architecture/trustedclusters/
2: https://goteleport.com/docs/management/admin/trustedclusters/

**Reply by pieter-lautus** [reply]
Thanks @webvictim 

We might consider an Enterprise license, the moderated SSH sessions feature looks great - ironically not because we are an enterprise, but because we are a small company needing to allow a small pool of developers do double duty as server support, but us also wanting to control sudo access.

I think what I'll do for now is keep the SSH daemon open, but with a locked-down config that only permits the "break glass" access and the scp we want to support. Self-hosted clusters may be in our future.
